msal/src/test/java/com/microsoft/identity/client/e2e/rules/RobolectricLoggingRule.java

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.client.e2e.rules;
+
+import android.util.Log;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.robolectric.shadows.ShadowLog;
+
+/**
+ * A JUnit rule to enable logging during a Robolectric Test.
+ */
+public class RobolectricLoggingRule implements TestRule {
```

Comment:
should we put this in common?

Reply:
Ideally yes, but unfortunately we can't put Robolectric as a direct dependency in common, whenever we tried that in the past, we ran into issues so for that reason we have to put Robolectric utilities directly in MSAL, and Broker repos.

Reply:
Should we create a new package that any of them can depend on, then?

If we can't do:
<pre>
MSAL -> Common (with Utilities) -> Robolectric
         ^
         |
Broker --+
</pre>

Can we make it:

<pre>
MSAL/Broker -> Common
            |
            +-> RobolectricTestUtils (or whatever)
</pre>